content/client-lib-development-guide/features.textile

```diff
@@ -81,7 +81,7 @@ h3(#restclient). RestClient
 *** @(RSC7b1)@ When it is not possible to send the @X-Ably-Lib@ header, such as for @JSONP@ requests, the library version should be sent as a query param such as @lib=js-1.0.0@
 ** @(RSC7c)@ If the @addRequestIds@ client option is enabled, every REST request to Ably should include in a @request_id@ query string parameter a random string obtained by url-safe base64-encoding a sequence of at least 9 bytes obtained from a source of randomness. This request ID must remain the same if a request is retried to a fallback host per @RSC15@. Any log messages associated with the request should include the request ID. If the request fails, the request ID must be included in the @ErrorInfo@ returned to the user.
 ** @(RSC7d)@ An @Agent@ library identifier is used to identify the library type and version, as well as relevant information describing any underlying layers and the client platform.
-*** @(RSC7d1)@ In general, the @Agent@ library identifier is composed of a series of @key/value@ entries joined by spaces as described in described by "RFC 7321 Section 5.5.3":https://tools.ietf.org/html/rfc7231#section-5.5.3. This must include at least the @<library name>/<library version>@ entry followed optionally, in dependency order, the @name/version@ entry for underlying library layer or platform layers. An example would be @ably-flutter/1.2.0 ably-java/1.2.1 android/24@. The documentation for each library must define the content and meaning of the @Agent@ identifiers advertised by that library.
+*** @(RSC7d1)@ In general, the @Agent@ library identifier is composed of a series of @key[/value]@ entries joined by spaces where @key@ is the name of a produce and @value@ is an optional version of the associated product which must be a valid "semantic version":https://semver.org/. This must include at least the @<library name>/<library version>@ entry followed optionally, in dependency order, the @name[/version]@ entry for underlying library layer or platform layers. An example would be @ably-flutter/1.2.0 ably-java/1.2.1 android/24@. The documentation for each library must define the content and meaning of the @Agent@ identifiers advertised by that library.
```

Comment:
```diff
- produce
+ product
```

Reply:
Also, why MUST it be a semantic version, rather than that being a recommendation?

Reply:
Also, why does this spec item start with "In general"? Is there a non-general equivalent?

Reply:
@lmars I fixed the spelling mistake, thanks. I also removed the "in general", I agree that it had no purpose there.

As for the SemVer requirement we've discussed this internally quite a bit. The general consensus has been that (1) It's a lot easier to relax this requirement later on than to start with a more liberal requirement and then make it more restrictive later on, and (2) If all of the versions are SemVer then this enforces consistency which may be useful later on if we need to parse this data - if we know everything will be SemVer then we can leverage the ton of already-existing open source SemVer parser out there rather than having to work with the versions on a case-by-case basis. I would be open to changing it to "should" but are there any particular reasons you have in mind for doing so?

Reply:
ah ok thanks, I formally remove my semver question 🙂